cfme/tests/infrastructure/test_providers.py

```diff
@@ -492,3 +495,56 @@ def test_infra_discovery_screen(appliance):
                    "to_ip4": ips['to']})
         view.start.click()
         view.flash.assert_message(ips['msg'])
+
+
+@pytest.fixture
+def setup_provider_min_templates(request, appliance, provider, min_templates):
+    templates_yaml = len(provider.data.get('templates', {}))
+    if templates_yaml < min_templates:
+        pytest.skip(f'Number of templates defined in yaml for {provider} does not meet minimum '
+                    f'for test parameter {min_templates}, skipping and not setting up provider')
+    if len(provider.mgmt.list_templates()) < min_templates:
+        pytest.skip(f'Number of templates on {provider} does not meet minimum '
+                    f'for test parameter {min_templates}, skipping and not setting up provider')
+    # Function-scoped fixture to set up a provider
+    setup_or_skip(request, provider)
+
+
+def verify_checked_items_compared(checkedList, view):
```

Comment:
TODO: move this method into the comparison view.

Reply:
Done as requested.